src/optionsProvider.js

```diff
@@ -13,18 +13,21 @@ module.exports = {
 	},
 	getKey: function(argv) {
 		return argv.key ||
-			process.env[argv.accesskeyvar] ||
+			process.env[argv.accesskey] ||
+			process.env.AWS_ACCESS_KEY_ID ||
```

Comment:
Can we just make the key stuff optional all the way through?  If we call the AWS SDK without specifying credentials we get environment variables, machine roles, and profiles all for free.
I'm a little bit concerned about how things will interact here -- if someone has environment variables including a session but then they pass credentials to the publisher explicitly it will pick up the explicit key and key id but it will still get the session token from the environment variable.

Reply:
FYI: I chatted a bit with @daryl-mcmillan off line! I will be remove some of the credentials checks and not specify the `aws env variables`, allowing the aws-sdk to pickup the credentials and handle errors by itself.